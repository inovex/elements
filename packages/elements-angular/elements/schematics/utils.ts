import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
