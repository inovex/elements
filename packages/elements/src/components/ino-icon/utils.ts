import { getAssetPath } from '@stencil/core';

let CACHED_MAP: Map<string, string>;

export function getIconMap(): Map<string, string> {
	if (!CACHED_MAP) {
		const win = window as any;
		win.inoIcons = win.inoIcons || {};
		CACHED_MAP = win.inoIcons.map = win.inoIcons.map || new Map();
	}
	return CACHED_MAP;
}

export function addIcons(icons: { [name: string]: string }) {
	const map = getIconMap();
	Object.keys(icons).forEach((name) => map.set(name, icons[name]));
}

export const getSrc = (src: string | undefined) => {
	if (isStr(src)) {
		src = src.trim();
		if (isSrc(src)) {
			return src;
		}
	}
	return null;
};

export function getName(name: string | undefined) {
	if (name && !isSrc(name)) {
		return name;
	}
	return undefined;
}

export function getUrl(src: string | undefined, icon: string | undefined) {
	let url = getSrc(src);

	if (url) {
		return url;
	}
	url = getName(icon);
	if (url) {
		return getNamedUrl(url);
	}

	url = getSrc(icon);
	if (url) {
		return url;
	}

	return null;
}

export function getNamedUrl(name: string) {
	const url = getIconMap().get(name);
	if (url) {
		return url;
	}
	return getAssetPath(`./ino-icon/${name}.svg`);
}

export const isSrc = (str: string) => str.length > 0 && /(\/|\.)/.test(str);

export const isStr = (val: any): val is string => typeof val === 'string';

export const toLower = (val: string) => val.toLowerCase();
