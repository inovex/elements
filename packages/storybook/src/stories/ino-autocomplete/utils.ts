import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
