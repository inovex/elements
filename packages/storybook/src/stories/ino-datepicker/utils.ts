import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
