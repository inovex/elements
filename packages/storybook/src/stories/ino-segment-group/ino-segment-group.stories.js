import componentReadme from '_local-elements/src/components/ino-segment-group/readme.md';
import withStencilReadme from '_local-storybookcore/with-stencil-readme';
import './ino-segment-group.scss';
import { boolean, select, text } from '@storybook/addon-knobs';

function subscribeToComponentEvents() {
  // == event block
  const eventHandler = function (e) {
    const el = e.target;
    if (el.tagName.toLowerCase() !== 'ino-segment-button') {
      return;
    }
    const group = el.closest('ino-segment-group');
    group.setAttribute('value', el.getAttribute('value'));
  };

  document.addEventListener('checkedChange', eventHandler);
  // == event block

  // unsubscribe function will be called by Storybook
  return () => {
    document.removeEventListener('checkedChange', eventHandler);
  };
}

export default {
  title: 'Buttons/<ino-segment-group>',

  decorators: [
    withStencilReadme(componentReadme),
    (story) => {
      subscribeToComponentEvents();
      return story();
    },
  ],
};

export const DefaultUsage = () => /* html */ `
    <div class="story-segment-group">
      <h4>Customizable group</h4>
      <div class="story-segment-group">
          <style>
            ino-segment-button.customizable-segment-btn {
                --ino-segment-button-color: ${text('--ino-segment-button-color', '#c1c1c1', 'Custom Properties')};
                --ino-segment-button-checked-color: ${text('--ino-segment-button-checked-color', '#0d10f3', 'Custom Properties')};
                --ino-segment-button-hover-color: ${text('--ino-segment-button-hover-color', '#5d60f7', 'Custom Properties')};
                --ino-segment-button-disabled-color: ${text('--ino-segment-button-disabled-color', '#777777', 'Custom Properties')};
                --ino-segment-button-hover-shadow: ${text('--ino-segment-button-hover-shadow', '0 4px 8px 0 rgba(61, 64, 245, 0.3), 0 12px 28px 0 rgba(61, 214, 245, 0.14)', 'Custom Properties')};
                --ino-segment-button-checked-shadow: ${text('--ino-segment-button-checked-shadow', '0 8px 16px 0 rgba(61, 64, 245, 0.3),0 16px 32px 0 rgba(61, 214, 245, 0.14)', 'Custom Properties')};
            }
          </style>
          <ino-segment-group id="segment-grp">
            <ino-segment-button
                value="1"
                class="customizable-segment-btn"
                disabled="${boolean('disabled', false)}"
                ino-dense="${boolean('ino-dense', false)}"
            >
            Option 1
            </ino-segment-button>
            <ino-segment-button
                value="2"
                class="customizable-segment-btn"
                disabled="${boolean('disabled', false)}"
                ino-dense="${boolean('ino-dense', false)}"
            >
            Option 2
            </ino-segment-button>
            <ino-segment-button
                value="3"
                class="customizable-segment-btn"
                disabled="${boolean('disabled', false)}"
                ino-dense="${boolean('ino-dense', false)}"
            >
            Option 3
            </ino-segment-button>
            <ino-segment-button
                value="4"
                class="customizable-segment-btn"
                disabled="${boolean('disabled', false)}"
                ino-dense="${boolean('ino-dense', false)}"
            >
            Option 4
            </ino-segment-button>
        </ino-segment-group>
      </div>
        <h4>Dense group with third option checked</h4>
        <ino-segment-group value="3" id="dense-segment-grp">
            <ino-segment-button ino-dense value="1">Option 1</ino-segment-button>
            <ino-segment-button ino-dense value="2">Option 2</ino-segment-button>
            <ino-segment-button ino-dense value="3">Option 3</ino-segment-button>
            <ino-segment-button ino-dense value="4">Option 4</ino-segment-button>
        </ino-segment-group>

        <h4>Group with no option checked</h4>
        <ino-segment-group id="segment-grp">
            <ino-segment-button value="1">Option 1</ino-segment-button>
            <ino-segment-button value="2">Option 2</ino-segment-button>
            <ino-segment-button value="3">Option 3</ino-segment-button>
            <ino-segment-button value="4">Option 4</ino-segment-button>
        </ino-segment-group>

        <h4>Group with a single button</h4>
        <ino-segment-group id="single-segment-grp">
            <ino-segment-button value="1">Option 1</ino-segment-button>
        </ino-segment-group>
    </div>
    `;
