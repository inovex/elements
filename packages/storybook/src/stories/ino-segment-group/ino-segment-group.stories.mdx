import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-segment-group/readme.md";

<Meta title="Buttons/ino-segment-group/Documentation" />

<Description markdown={readme} />
