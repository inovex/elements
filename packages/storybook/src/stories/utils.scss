/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
