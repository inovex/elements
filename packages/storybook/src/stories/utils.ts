import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
