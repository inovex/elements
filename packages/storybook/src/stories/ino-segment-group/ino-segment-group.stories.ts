import { useEffect } from '@storybook/client-api';
import { Components } from '@inovex.de/elements';
import { Story } from '@storybook/web-components';
import { html } from 'lit-html';
import { defaultDecorator } from '../utils';
import './ino-segment-group.scss';

const eventHandler = (e: Event) => {
  const el = e.target as HTMLElement;
  if (el.tagName.toLowerCase() !== 'ino-segment-button') {
    return;
  }
  const group = el.closest('ino-segment-group');
  group.setAttribute('value', el.getAttribute('value'));
};

export default {
  title: 'Buttons/ino-segment-group',
  component: 'ino-segment-group',
  decorators: [
    (story) => defaultDecorator(story, 'story-segment-group'),
    (story) => {
      useEffect(() => {
        document.addEventListener('checkedChange', eventHandler);
        return () =>
          document.removeEventListener('checkedChange', eventHandler);
      });
      return story();
    },
  ],
  parameters: {
    actions: {
      handles: ['checkedChange'],
    },
  },
};
export const Playground: Story<Components.InoSegmentGroup> = (args) => html`
  <ino-segment-group id="segment-grp" name="${args.name}" value="${args.value}">
    <ino-segment-button value="opt-1">Option 1</ino-segment-button>
    <ino-segment-button value="opt-2">Option 2</ino-segment-button>
    <ino-segment-button value="opt-3">Option 3</ino-segment-button>
    <ino-segment-button value="opt-4">Option 4</ino-segment-button>
  </ino-segment-group>
`;
Playground.args = {
  name: '',
  value: 'opt-2',
};
Playground.argTypes = {
  value: {
    control: {
      type: 'select',
    },
    options: ['opt-1', 'opt-2', 'opt-3', 'opt-4'],
  },
};

export const DenseGroupWithCheckedOption = () => html`
  <ino-segment-group value="3" id="dense-segment-grp">
    <ino-segment-button dense value="1">Option 1</ino-segment-button>
    <ino-segment-button dense value="2">Option 2</ino-segment-button>
    <ino-segment-button dense value="3">Option 3</ino-segment-button>
    <ino-segment-button dense value="4">Option 4</ino-segment-button>
  </ino-segment-group>
`;

export const GroupWithNoCheckedOption = () => html`
  <ino-segment-group id="segment-grp">
    <ino-segment-button value="1">Option 1</ino-segment-button>
    <ino-segment-button value="2">Option 2</ino-segment-button>
    <ino-segment-button value="3">Option 3</ino-segment-button>
    <ino-segment-button value="4">Option 4</ino-segment-button>
  </ino-segment-group>
`;

export const SingleButtonGroup = () => html`
  <ino-segment-group id="single-segment-grp">
    <ino-segment-button value="1">Option 1</ino-segment-button>
  </ino-segment-group>
`;

/*
<style>
ino-segment-button.customizable-segment-btn {
  --ino-segment-button-color: ${text(
    '--ino-segment-button-color',
    '#c1c1c1',
    'Custom Properties'
  )};
  --ino-segment-button-checked-color: ${text(
    '--ino-segment-button-checked-color',
    '#0d10f3',
    'Custom Properties'
  )};
  --ino-segment-button-hover-color: ${text(
    '--ino-segment-button-hover-color',
    '#5d60f7',
    'Custom Properties'
  )};
  --ino-segment-button-disabled-color: ${text(
    '--ino-segment-button-disabled-color',
    '#777777',
    'Custom Properties'
  )};
  --ino-segment-button-hover-shadow: ${text(
    '--ino-segment-button-hover-shadow',
    '0 4px 8px 0 rgba(61, 64, 245, 0.3), 0 12px 28px 0 rgba(61, 214, 245, 0.14)',
    'Custom Properties'
  )};
  --ino-segment-button-checked-shadow: ${text(
    '--ino-segment-button-checked-shadow',
    '0 8px 16px 0 rgba(61, 64, 245, 0.3),0 16px 32px 0 rgba(61, 214, 245, 0.14)',
    'Custom Properties'
  )};
}
</style>
*/
