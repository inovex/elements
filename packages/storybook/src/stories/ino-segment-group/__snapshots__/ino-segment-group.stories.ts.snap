// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons/ino-segment-group Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-segment-group">
    <ino-segment-group id="ino-segment-group-id__2s0h4k9dy"
                       name
                       value="opt-2"
                       class="hydrated"
    >
      <ino-segment-button value="opt-1"
                          class="hydrated"
      >
        Option 1
      </ino-segment-button>
      <ino-segment-button value="opt-2"
                          class="hydrated ino-segment-button--checked"
                          checked
      >
        Option 2
      </ino-segment-button>
      <ino-segment-button value="opt-3"
                          class="hydrated"
      >
        Option 3
      </ino-segment-button>
      <ino-segment-button value="opt-4"
                          class="hydrated"
      >
        Option 4
      </ino-segment-button>
    </ino-segment-group>
  </div>
</div>
`;
